components/homme/src/share/thread_mod.F90

```diff
@@ -16,18 +16,26 @@ module thread_mod
   implicit none
   private
 
-  integer, public :: NThreads   ! total number of threads
+#ifdef MODEL_CESM
+  integer, public, pointer :: NThreads   ! total number of threads
```

Comment:
Are these still needed?

Reply:
Somehow these variables are declared as pointers in CESM and I keep this block of declaration here to make sure that they can be initialized correctly in CESM.

Reply:
So CESM allocates these variables, and use them? Since they are nowhere used in E3SM, can maybe CESM keep these vars inside CAM's modules?

Reply:
Thanks @bartgol . You are right that CESM will allocate and use those variables. Since they share the same names (CESM declares them as pointer and E3SM declares them as scalar), I think I have to use the `ifdef` here to make sure only one version is used at runtime?

Reply:
Yes, I think that's appropriate. The code in EAM that uses homme's stuff will likely not like the pointer change...

Reply:
Thanks @bartgol . I will talk with the CAM developers and see if they can change the CAM's code to use scalar type for those variables instead. If it works, I can issue a separate pull request to remove the `ifdef` block here.